dotcom-rendering/src/web/components/Braze.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect } from 'react';
+import { ArticleDesign, storage } from '@guardian/libs';
+import { BrazeMessagesInterface } from '@guardian/braze-components';
+import {
+	incrementWeeklyArticleCount,
+	getWeeklyArticleHistory,
+} from '@guardian/support-dotcom-components';
+import { WeeklyArticleHistory } from '@guardian/support-dotcom-components/dist/dotcom/src/types';
+import { parse } from '@root/src/lib/slot-machine-flags';
+import { buildBrazeMessages } from '../lib/braze/buildBrazeMessages';
+import { SlotBodyEnd } from './SlotBodyEnd/SlotBodyEnd';
+import { StickyBottomBanner } from './StickyBottomBanner/StickyBottomBanner';
+import { hasOptedOutOfArticleCount } from '../lib/contributions';
+import { incrementDailyArticleCount } from '../lib/dailyArticleCount';
+import { useOnce } from '../lib/useOnce';
+
+type Props = {
+	idApiUrl: string;
+	contentType: string;
+	sectionName?: string;
+	shouldHideReaderRevenue: boolean;
+	isMinuteArticle: boolean;
+	isPaidContent: boolean;
+	tags: TagType[];
+	contributionsServiceUrl: string;
+	stage: string;
+	section: string;
+	isPreview: boolean;
+	isSensitive: boolean;
+	switches: Switches;
+	keywordIds: string;
+	pageId: string;
+	slotMachineFlags?: string;
+	format: ArticleFormat;
+};
+
+function shouldShowSlotBodyEnd(
+	format: ArticleFormat,
+	switches: Switches,
+	slotMachineFlags?: string,
+) {
+	if (!switches.slotBodyEnd) return false;
+	if (!parse(slotMachineFlags || '').showBodyEnd) return false;
+
+	switch (format.design) {
+		case ArticleDesign.Interactive:
+		case ArticleDesign.FullPageInteractive:
+			return false;
+		default:
+			return true;
+	}
+}
+
+export const Braze = ({
```

Comment:
Pulling this out of `App` looks great. I'm not sure `Braze` is the right name here, as `SlotBodyEnd` and `StickyBottomBanner` are broader than just Braze (they may also show epics/banners from RRCP and also the CMP banner). Not sure what a good name is that ties those two slots together though (maybe this is what `Automat` is 😄 )